client/dashboard/sites/settings-repositories/connect-repository-form.tsx

```diff
@@ -92,9 +92,21 @@ const RepositorySelector = ( {
 	field,
 	onChange,
 	data,
-}: DataFormControlProps< ConnectRepositoryFormData > ) => {
+	shouldRestoreFocus,
+}: DataFormControlProps< ConnectRepositoryFormData > & { shouldRestoreFocus: boolean } ) => {
 	const { id, getValue, description } = field;
 	const currentValue = getValue?.( { item: data } );
+	const comboboxRef = useRef< HTMLDivElement | null >( null );
+	useEffect( () => {
+		if ( shouldRestoreFocus && comboboxRef.current ) {
+			const input =
+				comboboxRef.current.querySelector< HTMLInputElement >( 'input[role="combobox"]' );
+
+			if ( input ) {
+				input.focus();
+			}
+		}
+	}, [ shouldRestoreFocus, currentValue, field.elements ] );
```

Comment:
I think it should be enough to keep `shouldRestoreFocus` only:

```suggestion
	}, [ shouldRestoreFocus ] );
```